rfcs/0086-create-secret.md

```diff
@@ -0,0 +1,59 @@
+---
+feature: New Syntax Create Secret
+authors:
+  - "TabVersion"
+start_date: "2024/03/18"
+---
+
+# Introducing SECRETs for Secure Storage of Sensitive Information
+
+## Background
+
+This RFC proposes a new syntax and implementation for securely storing and referencing sensitive information,
+such as database passwords and API keys, in SQL statements using a SECRET catalog. 
+The goal is to enhance the security of production environments by encrypting sensitive data and applying access controls.
```

Comment:
Could you elaborate the usage scenarios and explain how `SECRET` enhances security? And how is it access controlled?

Reply:
previously we don't have any control over the credentials. All sensitive info is exposed in plain text. Having a dedicated catalog is an improvement.

> And how is it access controlled?

For now, we do access control by enforcing the privileges set by meta, the same as other catalogs.

Reply:
> All sensitive info is exposed in plain text. Having a dedicated catalog is an improvement.

I mainly want to know what's the prior "exposed" cases that the new mechanism can solve. In other words, what's the threat model? (prevent who from accessing the secrets)

> we do access control by enforcing the privileges set by meta, the same as other catalogs

e.g., if we just want to prevent user B from accessing secrets that owned by user A, `CREATE SECRET` seems not useful, as they already don't have any privileges over tables.

Reply:
At the same time, we are also developing security features like redaction https://github.com/risingwavelabs/risingwave/pull/14310, so I hope these can be compared.

Reply:
After offline discussion, I now think it's sth like GHA secret:
- cannot be exposed
- can grant READ privilege to other users (i.e., use the secret without knowing the value)

So it makes more sense to me now. 

(And yes, redaction is not needed with this (if carefully implemented))